controller/tasks_manager.go

```diff
@@ -215,6 +215,42 @@ func (tm *TasksManager) TaskUpdate(ctx context.Context, updateConf config.TaskCo
 	return plan.ChangesPresent, plan.Plan, "", nil
 }
 
+// TaskFailSilently creates, runs, and adds a new task. It expects that this
+// task is highly unlikely to error because it has previously been created and
+// run before. Therefore it does not handle the error beyond logging.
+//
+// This method is used when we do not want the caller to error and exit when
+// creating, running, and adding a new task.
+func (tm *TasksManager) TaskFailSilently(ctx context.Context, taskConfig config.TaskConfig) {
```

Comment:
Was thinking that this would benefit from adding `Create` in the name, which then led me to think that it's technically `TaskCreateAndRunFailSilently` if we want parity with the other methods. So then on this note, wanted to get your thoughts on updating `TaskCreateAndRun` to have a boolean for whether errors are ignored or not instead of a separate method.

Also this is like a big nit, but if we're logging the errors, is it technically failing silently? 🤔 I think it gets the point across either way, so not really advocating a change!

Reply:
Thanks for the feedback!

I like having one methods with a boolean because it makes naming easier. I went with separate methods to reduce complexity within the method, to be more explicit about the two behaviors to avoid any accidents, to avoid having the API have access to the fail-silently option. 🤔let me try out combining again!

I get what you mean that "fail-silently" is not intuitive because the logs aren't very silent 😆 . From looking around initially, fail-silently seemed like it referred to only when errors are aren't returned/thrown (they may or may not be logged - logging seemed like a grey zone) ([ex](https://enterprisecraftsmanship.com/posts/fail-fast-principle/)). Since it's not intuitive, maybe "continue on failure" or "allow failure"?

Reply:
Hello! I tried out some iterations to combine the methods ([gist](https://gist.github.com/lornasong/247a0d96803c06bb8447ee183faa9ad1)). From that, I'm thinking of 2 options:
1. One combined method: iteration 2 in gist and use "allowFail" instead of "failSilently"
2. Two separate methods: with "allowFail" e.g. TaskCreateAndRunAllowFail

I think I'm leaning towards Op 1. Please let me know if you have any thoughts!

Reply:
Yeah, I think iteration 1 is what I was thinking of originally, but it's more difficult to read than anticipated! I'd be good with either of these two options! I might be feeling more the separate methods now since iteration 2 of the combined method doesn't really reduce code duplication, which I thought could have been the advantage of a combined method.

Reply:
Thanks for the feedback! I went with separate methods (op 2) and "allow fail" type of naming. Agree with your point about reducing code duplication is the advantage of the combined method

Reply:
Thanks for entertaining my original idea! Changes looks good to me!